js/map-content-to-topper.js

```diff
@@ -0,0 +1,162 @@
+		content.type === 'package' && isNews(content)) {
+
+		const designTheme = content.package && content.package.design.theme || content.design && content.design.theme;
+		const isStandaloneLiveBlog = !content.package && content.realtime && content.liveBlog;
+
+		const isLoud = designTheme === 'extra' || isStandaloneLiveBlog;
+
+		return {
+			layout: null,
+			backgroundColour: isLoud ? 'crimson' : 'wheat',
+			modifiers: ['news-package'],
+			themeImageRatio: 'split',
+			includesImage: true,
+			isExperimental: true,
+		};
+
+
+	//Articles within a package get a slate offset topper if the package has the 'extra' theme
+	} else if (content.containedIn && content.containedIn.length && content.package && content.package.design.theme.includes('extra') && !isNews(content.package)) {
+		return {
+			layout: 'full-bleed-offset',
+			largeHeadline: true,
+			backgroundColour: 'slate',
+			modifiers: ['full-bleed-offset', 'package-extra'],
+			includesImage: true
+		};
+
+	//package landing pages
+	} else if (content.type === 'package' && content.design && content.design.theme) {
+		const themeMap = {
+			'basic': {
+				bgColour: 'wheat',
+				layout: 'split-text-left',
+				largeHeadline: true
+			},
+			'special-report': {
+				bgColour: 'claret',
+				layout: 'split-text-left',
+				largeHeadline: true
+			},
+			'extra': {
+				bgColour: 'slate',
+				layout: 'split-text-left',
+				largeHeadline: true
+			},
+			'extra-wide': {
+				bgColour: 'slate',
+				layout: 'split-text-left',
+				largeHeadline: true
+			}
+		};
+
+		const selectedTheme = themeMap[content.design.theme];
+		const modifiers = [selectedTheme.layout, 'package', `package-${content.design.theme}`];
+
+		return {
+			layout: selectedTheme.layout,
+			largeHeadline: selectedTheme.largeHeadline,
+			backgroundColour: selectedTheme.bgColour,
+			modifiers,
+			includesImage: true
+		};
+
+		//otherwise use the editorially selected topper if it exists
+	} else if(content.topper && content.topper.layout && themeImageRatio.hasOwnProperty(content.topper.layout)) {
+		const hasImage = content.topper.layout !== 'full-bleed-text';
+		let backgroundColour;
+
+		//converts old palette colours to new palette colours from Methode
+		if (content.topper.layout === 'full-bleed-offset') {
+			backgroundColour = 'paper';
+		} else if (content.topper.backgroundColour === 'pink') {
+			backgroundColour = 'paper';
+		} else if (content.topper.backgroundColour === 'blue') {
+			backgroundColour = 'oxford';
+		} else {
+			backgroundColour = content.topper.backgroundColour || 'paper';
+		}
+
+		return {
+			layout: content.topper.layout,
+			largeHeadline: true,
+			backgroundColour,
+			modifiers: [content.topper.layout],
+			includesImage: hasImage
+		};
+
+		//Branded regular toppers
+	} else if(content.brandConcept || (content.genreConcept && content.genreConcept.id === 'e569e23b-0c3e-3d20-8ed0-4c17b8177c05')) {
```

Comment:
Hard-coded IDs in o-components doesn't feel right. IDs can change 😖 !

Reply:
what do you suggest here? looking at `prefLabel`?